src/Command/UpdateRecipesCommand.php

```diff
@@ -0,0 +1,407 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Flex\Command;
+
+use Composer\Command\BaseCommand;
+use Composer\IO\IOInterface;
+use Composer\Util\ProcessExecutor;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Formatter\OutputFormatterStyle;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Flex\Configurator;
+use Symfony\Flex\Downloader;
+use Symfony\Flex\Flex;
+use Symfony\Flex\GithubApi;
+use Symfony\Flex\InformationOperation;
+use Symfony\Flex\Lock;
+use Symfony\Flex\Recipe;
+use Symfony\Flex\Update\RecipePatcher;
+use Symfony\Flex\Update\RecipeUpdate;
+
+class UpdateRecipesCommand extends BaseCommand
+{
+    /** @var Flex */
+    private $flex;
+    private $downloader;
+    private $configurator;
+    private $rootDir;
+    private $githubApi;
+
+    public function __construct(/* cannot be type-hinted */ $flex, Downloader $downloader, $httpDownloader, Configurator $configurator, string $rootDir)
+    {
+        $this->flex = $flex;
+        $this->downloader = $downloader;
+        $this->configurator = $configurator;
+        $this->rootDir = $rootDir;
+        $this->githubApi = new GithubApi($httpDownloader);
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this->setName('symfony:recipes:update')
+            ->setAliases(['recipes:update'])
+            ->setDescription('Updates an already-installed recipe to the latest version.')
+            ->addArgument('package', InputArgument::OPTIONAL, 'Recipe that should be updated.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $win = '\\' === \DIRECTORY_SEPARATOR;
+        $runtimeExceptionClass = class_exists(RuntimeException::class) ? RuntimeException::class : \RuntimeException::class;
+        if (!@is_executable(strtok(exec($win ? 'where git' : 'command -v git'), \PHP_EOL))) {
+            throw new $runtimeExceptionClass('Cannot run "recipes:update": git not found.');
+        }
```

Comment:
I think it's better to use `ExecutableFinder` here and then use the found executable for all git commands? This allows best compatibility (e.g. when someone on Windows has `git.cmd` instead of `git`) and safety (e.g. when using this on a host with `open_basedir` set).

Reply:
That sounds good to me! But I copied this from `InstallRecipesCommand`, which was added by @nicolas-grekas https://github.com/symfony/flex/commit/d3adcac0fa1749403b4010f14af73438908c5d4b

Nicolas - did you purposely avoid `ExecutableFinder`?

Reply:
> e.g. when someone on Windows has git.cmd instead of git

Looking at Composer's code (which always uses `git`), I don't think we should care about this.

Here, we are checking if git is installed. `ExecutableFinder` might do the job for sure, but since we're relying on the OS discovery, the current logic is consistent to me.

No strong opinion.